packages/ember-meta/lib/meta.ts

```diff
@@ -208,55 +223,32 @@ export class Meta {
   }
 
   peekDeps(subkey: string, itemkey: string): number {
-    let pointer: Meta | undefined = this;
-    while (pointer !== undefined) {
-      let map = pointer._deps;
-      if (map !== undefined) {
-        let value = map[subkey];
-        if (value !== undefined) {
-          let itemvalue = value[itemkey];
-          if (itemvalue !== undefined) {
-            return itemvalue;
-          }
-        }
-      }
-      pointer = pointer.parent;
-    }
-
-    return 0;
+    let val = this._findInherited3('_deps', subkey, itemkey);
+    return val === undefined ? 0 : val;
   }
 
   hasDeps(subkey: string) {
-    let pointer: Meta | undefined = this;
-    while (pointer !== undefined) {
-      let deps = pointer._deps;
-      if (deps !== undefined && deps[subkey] !== undefined) {
-        return true;
-      }
-      pointer = pointer.parent;
-    }
-    return false;
+    let val = this._findInherited2('_deps', subkey);
+    return val !== undefined;
   }
 
   forEachInDeps(subkey: string, fn: Function) {
-    return this._forEachIn('_deps', subkey, fn);
-  }
-
-  _forEachIn(key: string, subkey: string, fn: Function) {
     let pointer: Meta | undefined = this;
     let seen: Set<any> | undefined;
     let calls: any[] | undefined;
     while (pointer !== undefined) {
-      let map = pointer[key];
+      let map = pointer._deps;
       if (map !== undefined) {
         let innerMap = map[subkey];
         if (innerMap !== undefined) {
           for (let innerKey in innerMap) {
             seen = seen === undefined ? new Set() : seen;
             if (!seen.has(innerKey)) {
               seen.add(innerKey);
-              calls = calls || [];
-              calls.push(innerKey, innerMap[innerKey]);
+              if (innerMap[innerKey] > 0) {
+                calls = calls || [];
+                calls.push(innerKey);
+              }
```

Comment:
pushing only dependencies which have count greater than zero, avoid checking it later downstream here: https://github.com/emberjs/ember.js/pull/16702/files#diff-8b1d0dc68f822472c1b8b7ae6148b656L168 and avoids calling callback function for deps count equal to zero